### PR TITLE
JetBrains: remove non-agent autocomplete and chat

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, the plugin had two code paths for autocomplete and chat: agent and non-agent. The non-agent path is no longer supported and can therefore be removed.

The motivation to remove this code is to simplify troubleshooting issues. Previously, the plugin would occasionally fallback to the non-agent code paths if the agent was not running. Now, we only try the agent instead.

This change makes it easier to troubleshoot issues since the logs will no longer include exceptions from non-agent code paths that were a red herring. This change also makes it easier to improve error handling since we don't have a fallback method we silently start using when the agent fails. Instead, we have to deal with the agent connection error immediately or report a helpful error message.

## Test plan

* Pull latest Cody changes
* `./gradlew -PforceAgentBuild=true :runIDE`
* Make sure autocomplete works
* Make sure sending a chat message works. Note that the embedding status is now hardcoded to an empty git repository.

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyEditorFactoryListener.java [hunk 1]
```diff
@@ -182,6 +182,9 @@ public static void informAgentAboutEditorChange(@Nullable Editor editor) {
     if (editor.getProject() == null) {
       return;
     }
+    if (!CodyAgent.isConnected(editor.getProject())) {
```
Comment (chwarwick) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyEditorFactoryListener.java:185]: This is resolving the regression I added there are 2 additional spots the `CodyAgentFocusListener` and `CodyFileEditorListener`

Comment (olafurpg) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyEditorFactoryListener.java:185]: Can you follow up with a PR to fix the other cases?

File: client/jetbrains/src/main/java/com/sourcegraph/cody/context/EmbeddingStatusView.java [hunk 2]
```diff
@@ -48,8 +43,7 @@ public EmbeddingStatusView(@NotNull Project project) {
     innerPanel.setBorder(new EmptyBorder(JBUI.insets(TEXT_MARGIN, TEXT_MARGIN, 0, TEXT_MARGIN)));
     this.add(innerPanel);
 
-    EdtExecutorService.getScheduledExecutorInstance()
-        .scheduleWithFixedDelay(this::updateEmbeddingStatusView, 1, 10, TimeUnit.SECONDS);
+    this.setEmbeddingStatus(new NoGitRepositoryEmbeddingStatus());
```
Comment (chwarwick) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/context/EmbeddingStatusView.java:46]: I think this is _probably_ ok for now but its the only indicator that the user is connected to the correct repo and embeddings status.  The debug logs will contain enough info so I think we'll still be able to debug issues.

Comment (olafurpg) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/context/EmbeddingStatusView.java:46]: We will need to fix this before we publish the plugin into the stable channel.

File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyEditorFactoryListener.java (hunk 1 above)
Comment (chwarwick) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyEditorFactoryListener.java:185]: https://github.com/sourcegraph/sourcegraph/pull/55999

**Review by chwarwick — APPROVED** [review]
I didn't go though all the removed code but the changes all look good

**Pull request merged**